tests/unit/libc_wrapper_test.cc

```diff
@@ -0,0 +1,55 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2024 ScyllaDB.
+ */
+#include <coroutine>
+#include <iostream>
+#include <grp.h>
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/reactor.hh>
+#include <seastar/core/seastar.hh>
+
+#include <seastar/testing/test_case.hh>
+
+using namespace seastar;
+
+SEASTAR_TEST_CASE(getgrnam_error_test) {
+    size_t buflen = 1;
+    char buf[1];
+    memset(&buf, 0, buflen);
```

Comment:
I don't think this memset is needed - I think getrnam() only writes to the buffer, it doesn't read it or expect it to be zeroed. But please correct me if I'm wrong.